### ISSUE TITLE
Boilerplate using redux-react?

### ISSUE THREAD

**Issue by trungpham** [makeomatic/redux-connect#39]
Does anyone know if there is any universal app boiler plate using this library?

**Comment by AVVS** [comment]
Havent seen one :(

**Comment by trungpham** [comment]
is there any lightweight webpack + redux + react-router that you can recommend? light weight enough that allows us to plug in redux-react?

**Comment by AVVS** [comment]
https://github.com/davezuko/react-redux-starter-kit is a pretty good one imo

**Comment by trungpham** [comment]
Checkout this project. They have a very clean way of prefetching data.
https://github.com/coodoo/react-redux-isomorphic-example

**Comment by AVVS** [comment]
Cant say anything except that I personally dont like it and at the time when I was writing this one - I looked through it - I dislike code style and I dislike having fetching utility in the boilerplate. The approach is rather similar in any case. 

For SSR you need to write a rendering function on the server, which accepts components from react-router matching function, traverses them, gathers functions that define how to fetch async props and form store, then you feed that to react renderer with a filled store and get what you need.

**Comment by trungpham** [comment]
Agree. I like your approach much better, avoiding mucking with the
component internal. Can we fork
https://github.com/davezuko/react-redux-starter-kit and add redux-connect
to it?

**Comment by AVVS** [comment]
I wont have time for this and there are implementations for it from previous `redux-async-connect`. Author of the template I've suggested is very receptive - so if you do write a PR - I believe it would be accepted

**Comment by trungpham** [comment]
This is another boilerplate project that has redux-async-connect integrated
already. There is an outstanding PR to move to redux-connect too.
https://github.com/erikras/react-redux-universal-hot-example/pull/1140

**Comment by AVVS** [comment]
do you want to add a PR to docs with that info?

**Comment by andrewBalekha** [comment]
> erikras/react-redux-universal-hot-example#1140

i'm using this one, but i've modified it to use your refactored version :)
so i can do that

**Comment by AVVS** [comment]
please do :)

**Comment by abhirathore2006** [comment]
@AVVS  i have few questions ? 

1. how this is different from original library ?

2. there are two ways described in original library , one was to use decorator and other to use static method, so does that static method still available in this library ? 

3. how static method is different from decorator and when to use which

**Comment by AVVS** [comment]
1. refactored to support nested dependant state loading, small syntax changed
2. only decorator, it's quite complex to define a user-provided static property, consult with code for more
3. static is a generated function based on this: https://github.com/makeomatic/redux-connect/blob/master/modules/containers/decorator.js

What decorator does is it creates a static function that is used when traversing resolved components for a given route in react-router. That's essentially the whole thing that this module does

**Comment by abhirathore2006** [comment]
@AVVS  thanks for clarification , also is it necessary to provide the decorator on Every component connected to router ?  will it be okay if i don't have decorator at root level component but have it on child components ?

**Comment by AVVS** [comment]
@abhirathore2006 decorators must be attached to components returned from react-router. Because pre-fetching is done _before_ render happens, it's essential that we gather information about state needed before the actual render. You can provide no decorators at all - then state wont be prefetched, you can provide just some decorators - then it will be fetched for the components that are returned for a given route. Hope I've answered your question

**Comment by abhirathore2006** [comment]
ok thanks :) @AVVS  , i will use this on my project and will let you know, how it goes
